File: Temp/Python/Practice/Runner.py
from Practice.Backtrack import NQueens
from Practice.Sort import InsertionSort, MergeSort, QuickSort

import numpy as np


class Runner(object):
    def __init__(self, *args):
        super(Runner, self).__init__(*args)

    @staticmethod
    def run_insertion_sort(arr):
        InsertionSort.get_instance(arr=arr).solve()

    @staticmethod
    def run_merge_sort(arr):
        MergeSort.get_instance(arr=arr).solve()

    @staticmethod
    def run_quick_sort(arr):
        QuickSort.get_instance(arr=arr).solve()

    @staticmethod
    def run_n_queens_backtrack(matrix: [[int], [int]]):
        NQueens.get_instance(matrix=matrix).solve()


def get_arr() -> [int]:
    return [20, 40, 10, 60, 20, 30, 50, 70, 80]


def get_matrix(dim: int) -> [[int], [int]]:
    return np.zeros(shape=(dim, dim))


def main(option=0):
    print(f"Running...")
    dimension: int = 8
    arr: [int] = get_arr()
    mtx: [[int], [int]] = get_matrix(dimension)
    runner: Runner = Runner()

    if option == 1:
        runner.run_insertion_sort(arr=arr)
    elif option == 2:
        runner.run_merge_sort(arr=arr)
    elif option == 3:
        runner.run_quick_sort(arr=arr)
    elif option == 4:
        runner.run_n_queens_backtrack(matrix=mtx)
    else:
        print("Invalid option")

    if 0 < option < 4:
        print(arr)
    elif option == 4:
        print(mtx)


if __name__ == '__main__':
    main(option=4)

File: Temp/Java/Practice/src/src/Runner.java
package src;


public class Runner {

	void printArr(int[] arr) {
		int N = arr.length;
		for (int i = 0; i < N; i++)
			System.out.printf(" %s ", arr[i]);
	}
	
	void printMtx(int[][] mtx) {
		int N = mtx.length;
		for (int row = 0; row < N; row++)
			for (int col = 0; col < N; col++)
				System.out.printf(" %s ", mtx[row][col] + (col == N - 1 ? "\n" : ""));
	}
	
	void execInsertionSort(int[] arr){
		sort.InsertionSort insertionSort = new sort.InsertionSort(arr);  
		insertionSort.solve();
		printArr(arr);
	}
	
	void execMergeSort(int[] arr) {
		sort.MergeSort mergeSort = new sort.MergeSort(arr);
		mergeSort.solve();
		printArr(arr);
	}
	
	void execQuickSort(int[] arr) {
		sort.QuickSort quickSort = new sort.QuickSort(arr);
		quickSort.solve();
		printArr(arr);
	}
	
	void execNQueens() {
		int N = 8;
		int[][] mtx = new int[N][N];
		backtrack.NQueens nQueens = new backtrack.NQueens(mtx);
		nQueens.solve();
		printMtx(mtx);
	}
	
	public static void main(String[] args) {
		Runner runner = new Runner();
		int[] arr = {20, 10, 40, 50, 60, 30, 70, 80};
		System.out.println("Running...");
		
		int option = 4;
		
		switch (option) {
		
			case 1: 
				runner.execInsertionSort(arr);
				break;
				
			case 2: 
				runner.execMergeSort(arr);
				break;
			
			case 3: 
				runner.execQuickSort(arr);
				break;
			
			case 4: 
				runner.execNQueens();
				break;
		
		default:
			throw new IllegalArgumentException("Unexpected value: " + option);
		}
			
		System.out.println("\nFinished...");
	}

}
